test/e2e/operator_test.go

```diff
@@ -226,6 +212,101 @@ func TestAWSLoadBalancerControllerWithDefaultIngressClass(t *testing.T) {
 	}
 }
 
+// TestAWSLoadBalancerControllerWithCredentialsSecret tests the basic happy flow for the operator
+// using the explicitly specified credentials secret.
+func TestAWSLoadBalancerControllerWithCredentialsSecret(t *testing.T) {
+	t.Log("Creating aws load balancer controller instance with credentials secret")
+
+	// provision AWS credentials for ALBC imitating the manually created secret
+	albcSecretName := "manual-aws-credentials-e2e"
+	albcSecretRef := corev1.ObjectReference{
+		Name:      albcSecretName,
+		Namespace: "aws-load-balancer-operator",
+	}
+	albcCrName := types.NamespacedName{
+		Name:      "mimic-manual-creds",
+		Namespace: "openshift-cloud-credential-operator",
+	}
+	albcCr, err := albcCredentialsRequest(albcCrName, albcSecretRef, "testsa")
+	if err != nil {
+		t.Fatalf("failed to generate desired credentials request to mimic manually created albc credentials: %v", err)
+	}
+	if err := kubeClient.Create(context.TODO(), albcCr); err != nil {
+		t.Fatalf("failed to create credentials request to mimic manually created albc credentials: %v", err)
+	}
+	defer func() {
+		waitForDeletion(t, kubeClient, albcCr, defaultTimeout)
+	}()
+
+	name := types.NamespacedName{Name: "cluster", Namespace: "aws-load-balancer-operator"}
+	alb := newALBCBuilder().withName(name).withCredSecret(albcSecretName).build()
+	if err := kubeClient.Create(context.TODO(), alb); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create aws load balancer controller %q: %v", name, err)
+	}
+	defer func() {
+		waitForDeletion(t, kubeClient, alb, defaultTimeout)
+	}()
+
+	expected := []appsv1.DeploymentCondition{
+		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+	}
+	deploymentName := types.NamespacedName{Name: "aws-load-balancer-controller-cluster", Namespace: "aws-load-balancer-operator"}
+	if err := waitForDeploymentStatusCondition(t, kubeClient, defaultTimeout, deploymentName, expected...); err != nil {
+		t.Fatalf("did not get expected available condition for deployment: %v", err)
+	}
+
+	// verify that the secret used by ALBC is different from the default provisioned by the operator
+	if verified, err := verifyConsumedCredentialsSecret(kubeClient, deploymentName, albcSecretName); err != nil {
+		t.Fatalf("failed to verify consumed credentials secret for controller deployment: %v", err)
+	} else if !verified {
+		t.Fatalf("expected credentials secret %q is not consumed by the controller deployment %q", albcSecretName, deploymentName)
+	}
+
+	testWorkloadNamespace := "aws-load-balancer-test-default-ing"
+	echoSvc := createTestWorkload(t, testWorkloadNamespace)
```

Comment:
Do we need a deferred cleanup step for the test workload?

Reply:
Each test workload is created in its own namespace. Then the ingress is created in this namespace. The ingress is set to be removed with a deferred function. So, the workload is left untouched because it cannot influence the work of ALBC.
Btw I copied this test case from the other one and forgot to change the namespace - addressing this.

Reply:
Hm, so each E2E test that creates an ingress _does_ clean up the ingress because it could impact other E2E tests but _doesn't_ clean up the workload namespace, pod, and service because they are harmless litter?  And tests are idempotent (you can run the full test suite multiple times on a cluster) because `createTestWorkload` ignores already-exists errors, right?  

I generally prefer that each test clean up all its resources so that it leaves the cluster as close as possible to its initial state when the test finishes.  However, we can follow the prevailing style of `operator_test.go` for this PR at least.